shared/reducers/fs.js

```diff
@@ -0,0 +1,21 @@
+// @flow
+import * as FSGen from '../actions/fs-gen'
+import * as Constants from '../constants/fs'
+import * as Types from '../constants/types/fs'
+
+const initialState = Constants.makeState()
+
+export default function(state: Types.State = initialState, action: FSGen.Actions) {
+  switch (action.type) {
+    case FSGen.resetStore:
+      return initialState
+    case FSGen.increaseCount: {
+      const {amount = 1} = action.payload
+      return state.withMutations(s => s.set('counter', state.counter + amount))
+    }
+    default:
+      // eslint-disable-next-line no-unused-expressions
+      ;(action: empty) // if you get a flow error here it means there's an action you claim to handle but didn't
```

Comment:
Same--what does this syntax do?

Reply:
this is basically to get flow to help us ensure we handle all cases in the switch statement.

ala: https://github.com/facebook/flow/issues/2695#issuecomment-256845109

this is a force cast of the action to a type that doesn't exist so it'll complain. try and comment out a case and run flow and it'll complain
`yarn flow`

Reply:
In the examples there's no `;`, is that needed?

Reply:
prettier is injecting that in automatically

Reply:
I see, thanks.

Reply:
Cool hack! Why is the semicolon needed here? I see linked GH comment doesn't have it.

Reply:
Oops nvm just say jzila asked the same question above.